src/components/AddItemModal/AddItemModal.tsx

```diff
@@ -0,0 +1,250 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Modal, Box, Typography, Select, MenuItem, TextField, Button, Autocomplete, AutocompleteChangeDetails, AutocompleteChangeReason } from '@mui/material';
+import { useState } from 'react';
+
+type InventoryItem = {
+  id: number;
+  name: string;
+  type: string;
+  quantity: number;
+  category: string;
+  description: string;
+  status: string;
+};
+
+type FormData = {
+  name: string;
+  type: string;
+  quantity: number;
+  category: string;
+  description: string;
+};
+
+type AddItemModalProps = {
+  addModal: boolean;
+  handleAddClose: () => void;
+  fetchData: () => void;
+  uniqueCategories: string[];
+  originalData: InventoryItem[];
+}
+
+const API = "/data-api/rest/item";
+const HEADERS = { 'Accept': 'application/json', 'Content-Type': 'application/json;charset=utf-8', }; //The server is denying me access to post. In the swa config file, if I add 'create' to the anonymous role, it will grant access. Needs to be updated later.
```

Comment:
```suggestion
const HEADERS = { 'Accept': 'application/json', 'Content-Type': 'application/json;charset=utf-8', }; //TODO The server is denying me access to post. In the swa config file, if I add 'create' to the anonymous role, it will grant access. Needs to be updated later.
```